src/hipoc/hipoc_program.cpp

```diff
@@ -59,34 +66,73 @@ namespace miopen {
 #if !MIOPEN_USE_COMGR
 namespace {
 
-inline std::string GetCodeObjectVersionOptionImpl()
+int DetectCodeObjectOptionSyntax()
 {
-    auto code_object_version = miopen::Value(MIOPEN_DEBUG_OPENCL_ENFORCE_CODE_OBJECT_VERSION{});
+    auto syntax = miopen::Value(MIOPEN_DEBUG_OPENCL_ENFORCE_CODE_OBJECT_OPTION{});
+    if(syntax > 4)
+    {
+        MIOPEN_LOG_E("Bad MIOPEN_DEBUG_OPENCL_ENFORCE_CODE_OBJECT_OPTION, using default");
+        syntax = 0;
+    }
+
+    if(syntax == 0)
+    {
+        if(HipCompilerVersion() >= external_tool_version_t{4, 0, 999999}) /// \ todo
```

Comment:
This should be updated when the compiler will accept "-mcode-object-version = N".